### PR TITLE
Fabo/1797 cookie bar

### LINKED ISSUE #1797: Define cookie policy 🍪
Description:

We need to define this on Voyager web for GDPR compliance

### PR DESCRIPTION
**Pull request by faboweb**
Closes #1797 

Looks like this right now:
image

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `PENDING.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
Missing actual terms of service

File: app/src/renderer/components/common/CookieBar.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div v-if="!session.cookiesAccepted" class="cookieBar">
+    <a class="close">
```
Comment (fedekunze) [inline app/src/renderer/components/common/CookieBar.vue]: why anchor ?

File: app/src/renderer/components/common/CookieBar.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      <i class="material-icons" @click="accept">close</i>
+    </a>
+    Lunie uses cookies and remote error collection to improve the app.
+    By dismissing this notification, you consent to this policy.
+    <a href="/privacy-policy">Click to learn more.</a>
```
Comment (fedekunze) [inline app/src/renderer/components/common/CookieBar.vue]: use vue-router

File: app/src/renderer/vuex/modules/session.js [hunk 3]
```diff
@@ -140,7 +144,7 @@ export default () => {
     // TODO split into sign in with ledger and signin with local key
     async signIn(
       { state, commit, dispatch },
-      { localKeyPairName, address, sessionType = `local`, errorCollection = false }
+      { localKeyPairName, address, sessionType = `local` }
```
Comment (fedekunze) [inline app/src/renderer/vuex/modules/session.js]:
```suggestion
      { localKeyPairName, address, sessionType = `ledger` }
```

File: app/src/renderer/components/common/CookieBar.vue (hunk 2 above)
Comment (fedekunze) [inline app/src/renderer/components/common/CookieBar.vue]: Missing actual `private-policy` and route definition

File: app/src/renderer/components/common/CookieBar.vue [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div v-if="!session.cookiesAccepted" class="cookieBar">
+    <a class="close">
+      <i class="material-icons" @click="accept">close</i>
+    </a>
+    Lunie uses cookies and remote error collection to improve the app.
+    By dismissing this notification, you consent to this policy.
```
Comment (fedekunze) [inline app/src/renderer/components/common/CookieBar.vue]: does this comply to GDPR ?

File: app/src/renderer/components/common/TmSessionSignUp.vue [hunk 5]
```diff
@@ -131,25 +131,6 @@
...
-          <div class="tm-field-checkbox">
-            <div class="tm-field-checkbox-input">
-              <input
-                id="error-collection"
-                v-model="fields.errorCollection"
-                type="checkbox"
-              >
-            </div>
-            <label class="tm-field-checkbox-label" for="error-collection">
-              I'd like to opt in for remote error tracking to help improve
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmSessionSignUp.vue]: why are we removing the checkbox ?

File: app/src/renderer/vuex/modules/session.js [hunk 6]
```diff
@@ -186,41 +185,58 @@ export default () => {
...
       commit(`setUserAddress`, null)
     },
-    loadErrorCollection({ state, dispatch }, address) {
-      const errorCollection =
-        localStorage.getItem(`${ERROR_COLLECTION_KEY}_${address}`) === `true`
+    loadLocalPreferences({ state, dispatch }) {
+      const localPreferences = localStorage.getItem(USER_PREFERENCES_KEY)
+
+      if (!localPreferences) {
+        state.cookiesAccepted = false
```
Comment (fedekunze) [inline app/src/renderer/vuex/modules/session.js]: use `commit`

File: app/src/renderer/vuex/modules/session.js [hunk 7]
```diff
@@ -186,41 +185,58 @@ export default () => {
...
-        state.externals.track(`pageview`, {
-          dl: window.location.pathname
-        })
       } else {
         console.log(`Error collection has been disabled`)
         state.externals.Sentry.init({})
       }
+    },
+    setAnalyticsCollection({ state, dispatch }, enabled) {
+      state.analyticsCollection = enabled
```
Comment (fedekunze) [inline app/src/renderer/vuex/modules/session.js]: ditto

File: app/src/renderer/components/common/TmSessionSignUp.vue (hunk 5 above)
Comment (faboweb) [reply app/src/renderer/components/common/TmSessionSignUp.vue]: I intended to simplify the cookie process for the user. We are still transparent about what happens if the user accepts it.

File: app/src/renderer/components/common/CookieBar.vue (hunk 4 above)
Comment (faboweb) [reply app/src/renderer/components/common/CookieBar.vue]: As far as I read yes. The user just needs to do an opt in action.

File: app/src/renderer/components/common/CookieBar.vue (hunk 1 above)
Comment (faboweb) [reply app/src/renderer/components/common/CookieBar.vue]: what would you use?

File: app/src/renderer/components/common/CookieBar.vue (hunk 2 above)
Comment (faboweb) [reply app/src/renderer/components/common/CookieBar.vue]: I will add that when we have the actual policy

File: app/src/renderer/vuex/modules/session.js (hunk 6 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/session.js]: personally I have the feeling this just generates more code as the mutation is just setting this value nothing else. why do you think a commit here is valuable?

Comment (fedekunze) [reply app/src/renderer/vuex/modules/session.js]: oh I was just following our coding standards, but happy to discuss it with the team

File: src/App.vue [hunk 8]
```diff
@@ -1,19 +1,21 @@
 <template>
-  <div id="app">
-    <Session />
-    <template>
+  <div>
+    <CookieBar />
```
Comment (jbibla) [inline src/App.vue]: @faboweb i don't know about this lint rule 🐫 for vue components...

Comment (faboweb) [reply src/App.vue]: We can change it, but I don't think it is so bad.

Comment (jbibla) [reply src/App.vue]: whatever the strongest vue recommendation is...

**Pull request merged**